frontend/packages/console-app/src/components/detect-perspective/DetectPerspective.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { connect, Dispatch } from 'react-redux';
+import { useExtensions, isPerspective } from '@console/plugin-sdk';
+import { getActivePerspective } from '@console/internal/reducers/ui';
+import { RootState } from '@console/internal/redux';
+import * as UIActions from '@console/internal/actions/ui';
+
+type OwnProps = {
+  children: React.ReactNode;
+};
+
+type StateProps = {
+  activePerspective: string;
+};
+
+type DispatchProps = {
+  setActivePerspective: (string) => void;
+};
+
+type DetectPerspectiveProps = OwnProps & StateProps & DispatchProps;
+
+const DetectPerspective: React.FC<DetectPerspectiveProps> = ({
+  activePerspective,
+  children,
+  setActivePerspective,
+}) => {
+  let detectedPerspective: string;
+  let detectionComplete: boolean;
+  const perspectiveExtensions = useExtensions(isPerspective);
+  const defaultPerspective = perspectiveExtensions.find((p) => p.properties.default);
+  const perspectiveDetectors = perspectiveExtensions.filter(
+    (p) => p.properties.usePerspectiveDetection,
+  );
+
+  perspectiveDetectors.some((p) => {
+    const isDetectedPerspective = p.properties.usePerspectiveDetection();
```

Comment:
Even if `activePerspective` is present, detection is being run. This should not happen.

This can be avoided by conditionally rendering a component which does the detection otherwise rendering the children. This conditional detection component will result in calling a redux action which will round trip and render the app children instead of the detection component.

Reply:
Updated.